pyvista/core/datasetattributes.py

```diff
@@ -288,7 +291,12 @@ def pop(self, key):
             copy = vtk_arr.NewInstance()
             copy.DeepCopy(vtk_arr)
             vtk_arr = copy
-        self.remove(key)
+        try:
+            self.remove(key)
+        except KeyError:
+            if default in self.pop.__defaults__:
+                raise KeyError
```

Comment:
I don't think you need the `KeyError`, the bare `raise` will raise the previous error which is already a `KeyError` (and might contain more useful information about what the problematic key was)

Reply:
remove() can raise KeyError, through get_array(), and raise in an except re-raises the exception.

Reply:
I think we're saying the same thing. I'm suggesting that 70174ee should be reverted

Reply:
Agreed.  Sorry, my bad suggestion!